synapse/lib/ast.py

```diff
@@ -743,6 +743,8 @@ async def genx():
             async for item in scmd.execStormCmd(runt, genr):
                 yield item
 
+            await genr.aclose()
```

Comment:
Let's put this in a finally block (with the try at line 738) so we guarantee cleanup.